src/control/events/ras.go

```diff
@@ -55,6 +55,8 @@ const (
 	RASSystemStopFailed        RASID = C.RAS_SYSTEM_STOP_FAILED         // error
 	RASEngineJoinFailed        RASID = C.RAS_ENGINE_JOIN_FAILED         // error
 	RASSystemFabricProvChanged RASID = C.RAS_SYSTEM_FABRIC_PROV_CHANGED // info
+	RASNVMeLinkSpeedDown       RASID = C.RAS_DEVICE_LINK_SPEED_DOWN     // error
+	RASNVMeLinkWidthDown       RASID = C.RAS_DEVICE_LINK_WIDTH_DOWN     // error
```

Comment:
I think it would be better to just have "Changed" events, as it's also relevant if one of these values is upgraded, right? Imagine there is a flaky bit of hardware that keeps upgrading/downgrading (maybe it needs to be re-seated or something).

With the current approach, the admin would only see the following:
  * 01:02:03 $device speed downgraded to 1Gbps
  * 01:02:05 $device speed downgraded to 1Gbps
  * 01:02:07 $device speed downgraded to 1Gbps
  
Whereas if these are changed events, the picture is a little clearer IMO:
  * 01:02:03 $device speed changed to 1Gbps
  * 01:02:04 $device speed changed to 10Gbps
  * 01:02:05 $device speed changed to 1Gbps
  * 01:02:06 $device speed changed to 10Gbps  
  * 01:02:07 $device speed changed to 1Gbps

Reply:
Agreed, the detailed can just be included in the message supplied to `NewGenericEvent()` with ID of a "Changed" event.

Reply:
Although there currently isn't a mechanism of storing the "downgraded" state, so the only reportable event, on health scan, is "negotiated != max". This single event type covers the original requirement for this change (DAOS-15967) and as I am not planning on offering to implement polling on link state (which was one of the proposals/requests) from either the engine or the control-plane side, it's likely that there will be some deployment specific polling of health stats via dmg commands.

That said I think you correct and the right approach is to provide changed events from any previous state. I propose to implement a `GetLastHealthStats()` engine interface method which we can use to implement the changed events. This result in a more sensible set of published events. Thoughts?

Reply:
Implemented: changed events by referring to "last" health stats recorded which are access through Engine interface